src/components/DateTimePicker/index.jsx

```diff
@@ -26,32 +26,72 @@ const DateTimePicker = ({
   value,
   labelProps,
   required = false,
-  ...otherProps
+  id,
+  onDatePickerBlur = noop,
+  onTimePickerBlur = noop,
+  onDatePickerFocus = noop,
+  onTimePickerFocus = noop,
+  datePickerProps,
+  timePickerProps,
+  autoUpdateTime = true,
 }) => {
   const [open, setOpen] = React.useState(false);
   const [date, setDate] = useState();
   const [time, setTime] = useState();
+  const [isHourFocused, setIsHourFocused] = useState(false);
+  const [isMinuteFocused, setIsMinuteFocused] = useState(false);
 
   const timeRef = React.useRef(null);
-  const id = useId(otherProps.id);
-  const errorId = `error_${id}`;
+  const defaultId = useId(id);
+  const errorId = `error_${defaultId}`;
   const handleDateChange = date => {
     setDate(date);
-    setTime(date);
+    if (autoUpdateTime) setTime(date);
     setOpen(false);
     onChange(date);
+    onDatePickerBlur(date);
```

Comment:
the onBlur is not being executed when we click on a date from the popup, the popup closes but the onBlur event is not executed when we close the popup.

Reply:
Then the name of the function doesn't make sense. It will be confusing to the developer since it won't be called when the date picker input loses focus.

Before diving into this further, can we make sure whether we need these additional callbacks `onDatePickerBlur`, `onTimePickerBlur`, `onDatePickerFocus`, `onTimePickerFocus` etc.? Can't we use the `onChange` handler of the component to save the values?